src/meta/src/barrier/context/recovery.rs

```diff
@@ -56,6 +57,8 @@ impl GlobalBarrierWorkerContextImpl {
             .reset_all_refresh_jobs_to_idle()
             .await?;
 
+        self.abort_dirty_pending_sink_state(database_id).await?;
```

Comment:
I think we should ensure that all sink coordinators of the database has stopped before we call this method, and otherwise there can be new rows written concurrently. But in current code path, we call sink_manager.reset() after reload runtime info, which is later than calling this method. Besides, we may need support resetting the coordinators of specific database on database recovery.